### DISCUSSION TITLE
Listeners Requests api queries params

### DISCUSSION THREAD

**Discussion by yarik2720** [AzuraCast/AzuraCast#4481] [Q&A] [answered]
Is it possible to add in api ability to sort order by such values as "Modified Date" from new to oldest?
Also is it possible to make pagination in requests to api because there is a lot of tracks?

**Comment by Vaalyn** [comment, marked as answer]
You can use pagination via the `per_page` and `page` parameters like this: `/api/station/1/requests?per_page=5&page=2`

The fields that can currently be used to sort the result are defined here: https://github.com/AzuraCast/AzuraCast/blob/main/src/Controller/Api/Stations/RequestsController.php#L77-L81

If you'd like to add fields to this list you could sent us a Pull Request with them added there.
The array consists of `<parameter_name>` => `<entity.column>` definitions.

The fields/columns that are currently in that list have an index defined in the DB to improve the query performance so when sorting by newly added columns with a large media library this can be slow / resource intensive. Wanted to mention this beforehand.

**Reply by yarik2720** [reply]
I can see, that for web requests there is added search ability by name, Is that possible to add also in api to get requests?

**Reply by Vaalyn** [reply]
This is already possible via the `searchPhrase` parameter although the response format will be a bit different when using it. Also for pagination you'll have to use `rowCount` instead of `per_page` and `current` instead of `page` when using the search.

**Comment by bugoverfl0w** [comment]
Please update your api docs bro :| many apis that do not update latest docs
